### PR TITLE
snap,client: use a different exit code for retryable errors

### PR DESCRIPTION
**Pull request by mvo5**
Some errors (like change-conflict) are not fatal and can be
retried later. We ot a request from users to return a different
error code in this case.

### CODE REVIEW

File: cmd/snap/main.go [hunk 1]
```diff
@@ -418,6 +418,9 @@ func main() {
 	// no magic /o\
 	if err := run(); err != nil {
 		fmt.Fprintf(Stderr, errorPrefix, err)
+		if client.IsRetryableError(err) {
+			os.Exit(10)
```
Comment (zyga) [inline cmd/snap/main.go]: Is the exit code significant?

File: tests/main/retryable-error/task.yaml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+execute: |
+    echo "Install a snap"
+    snap install test-snapd-tools &
+
+    echo "And try to install it again which results in a change confict error"
+    while true; do
+        if snap changes |grep "Doing.*Install"; then
+            if snap install test-snapd-tools; then
+                echo "snap install should retrun a change-conflict: test brokwn"
```
Comment (zyga) [inline tests/main/retryable-error/task.yaml]: Typo: "broken" (ironic :)

File: cmd/snap/main.go (hunk 1 above)
Comment (chipaca) [reply cmd/snap/main.go]: I would've gone with 9 (because it's http 409) :-)

File: tests/main/retryable-error/task.yaml (hunk 2 above)
Comment (chipaca) [inline tests/main/retryable-error/task.yaml]:
```suggestion
                echo "snap install should return a change-conflict: test broken"
```

Comment (mvo5) [reply tests/main/retryable-error/task.yaml]: \o/

File: client/client.go [hunk 3]
```diff
@@ -445,6 +445,16 @@ const (
 	ErrorKindDaemonRestart = "daemon-restart"
 )
 
+// IsRetryableError returns true if the given error is an error
+// that can be retried later.
+func IsRetryableError(err error) bool {
```
Comment (pedronis) [inline client/client.go]: wonder if this should drop the "Error" to be more consistent with stdlib names like this, IsRetryable ?

**Pull request merged**